.github/workflows/check.yaml

```diff
@@ -5,6 +5,27 @@ on:
   - pull_request
 
 jobs:
+  pre-commit:
+    name: pre-commit
+    runs-on: ubuntu-latest
+    if: ${{ !startsWith(github.ref, 'refs/tags') }}
+
+    steps:
+      - uses: actions/checkout@v6
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v6
+        with:
+          python-version: "3.11"  # minimum supported lang version
+
+      - name: Install dependencies
+        run: python -m pip install hatch 'click!=8.3.0'
+
+      - name: Run pre-commit hooks
+        run: hatch run lint:install-hooks && hatch run lint:check
```

Comment:
I think you meant ` run: hatch run lint:precommit` instead of `hatch run lint:check`